mattermost/mattermost-webapp
# Cypress/E2E: Fix tests for cloud (part 4)

#### Summary
Fix tests for cloud.


1. Test metadata
    - Add `@not_cloud` and `@te_only` labels to easily sort out spec files and run against corresponding test server
2. Flags to fail fast
      - Add `cy.shouldNotRunOnCloudEdition`, `cy.shouldRunOnTeamEdition`, `cy.shouldHaveElasticsearchDisabled` and `cy.shouldHavePluginUploadEnabled` to fail fast whenever certain requirement is not met
3. Default config for on prem and cloud edition
      - Rename `partial_default_config.json` to `on_prem_default_config.json`, and move from `/fixtures` to `/support/api` folder 
      - Add `cloud_default_config.json` to easily identify what are the config that can be update by Cypress against Cloud server
      - Add `expectConfigToBeUpdatable` to fail whenever a setting can't be updated which normally happened with Cloud server where not all config can be updated by a sysadmin.
4. Move spec files
    - move tests into its own files if it has specific requirements. Most of those are tests that can't be ran on Cloud, or only for team edition
5. Remove `cy.dbGetUser` in the global `before` hook which tends to throw an error connection when connecting/disconnecting for each spec files
6. Remove unnecessary update to config when particular setting is already in the default config.
7. Remove `cy.apiDeleteLicense` whenever a test requires no license (or simply TE only). Instead, it's replaced with flag `cy.shouldRunOnTeamEdition` to check whether a server is on team edition. The idea is to not dynamically add or remove license when running Cypress test but make the test server license status to remain as is all through out the entire test. With this we'll have separate test run for TE, Enterprise (or E20) and Cloud. Also, note that in Cloud, we should not delete the license.
8. Fix some typo, ID and API return values.

To follow: Will update Cypress E2E docs

Submitted on top of https://github.com/mattermost/mattermost-webapp/pull/7143
